spec/unit/shared_examples/volume_attachments.rb

```diff
@@ -0,0 +1,94 @@
+    allow_any_instance_of(target_class).to receive(:[]).with('name')
+    allow_any_instance_of(target_class).to receive(:exists?).and_return(false)
+    allow_any_instance_of(target_class).to receive(:create).and_return(true)
+    allow_any_instance_of(base_sdk::StorageSystem).to receive(:retrieve!).and_return(true)
+    allow_any_instance_of(base_sdk::StoragePool).to receive(:retrieve!).and_return(true)
+    expect_any_instance_of(target_provider_class)
+      .to receive(:load_resource)
+      .with(:StorageSystem, { hostname: 'StorageSystem1', name: 'StorageSystem1' }, :uri)
+      .and_call_original
+    expect_any_instance_of(target_provider_class)
+      .to receive(:load_resource)
+      .with(:StorageSystem, { hostname: 'StorageSystem2', name: 'StorageSystem2' }, :uri)
+      .and_call_original
+    expect(base_sdk::StorageSystem).to receive(:new)
+      .and_return(base_sdk::StorageSystem.new(client, uri: 'fake/StorageSystem1'), base_sdk::StorageSystem.new(client, uri: 'fake/StorageSystem2'))
+    expect_any_instance_of(target_provider_class)
+      .to receive(:load_resource)
+      .with(:StoragePool, name: 'StoragePool1', storageSystemUri: 'fake/StorageSystem1')
+      .and_call_original
+    expect_any_instance_of(target_provider_class)
+      .to receive(:load_resource)
+      .with(:StoragePool, name: 'StoragePool2', storageSystemUri: 'fake/StorageSystem2')
+      .and_call_original
+    expect_any_instance_of(target_class).to receive(:create_volume_with_attachment)
+      .with(instance_of(base_sdk::StoragePool), { 'name' => 'Volume1' }, 'attr_1' => 'attr 1')
+    expect_any_instance_of(target_class).to receive(:create_volume_with_attachment)
+      .with(instance_of(base_sdk::StoragePool), { 'name' => 'Volume2' }, 'attr_2' => 'attr 2')
+    expect_any_instance_of(target_class).to receive(:[]).twice.with('sanStorage').and_return({})
+    expect_any_instance_of(base_sdk::Volume).not_to receive(:retrieve!)
+    expect_any_instance_of(target_class).not_to receive(:add_volume_attachment)
+    expect(real_chef_run).to public_send(*target_match_method)
+  end
+end
+
+# NOTE:
+# This shared example requires the following variables:
+#  target_class - Full name of the OneView resource to be tested
+#  target_match_method - Array with name of match method called and with the argument of the match method,
+#    e.g: let(:target_match_method) { [:create_oneview_server_profile, 'ServerProfileName'] }
+RSpec.shared_examples 'action :create #create_volume_attachments with wrong data' do
+  it 'should #add_volume_attachment' do
+    allow_any_instance_of(target_class).to receive(:exists?).and_return(false)
+    expect_any_instance_of(target_class).not_to receive(:create_volume_with_attachment)
+    expect_any_instance_of(target_class).not_to receive(:add_volume_attachment)
+    expect { real_chef_run }.to raise_error(/To add volume attachments you need to specify the 'volume' or 'volume_data' insides 'volume_attachments' options/)
```

Comment:
If you are using a matcher just match the relevant part of the error message, this way you won't need to change the test if the message in the code is changed (That is the case here)